usaspending_api/search/v2/views/spending_by_award.py

```diff
@@ -226,3 +280,122 @@ def populate_response(self, results: list, has_next: bool) -> dict:
             "page_metadata": {"page": self.pagination["page"], "hasNext": has_next},
             "messages": [get_time_period_message()],
         }
+
+    def populate_elastic_response(
+        self, results: list, has_next: bool, last_id: str = None, last_value: str = None
+    ) -> dict:
+        return {
+            "limit": self.pagination["limit"],
+            "results": results,
+            "page_metadata": {"page": self.pagination["page"], "hasNext": has_next},
+            "last_id": last_id,
+            "last_value": last_value,
+            "messages": [get_time_period_message()],
+        }
+
+    def query_elasticsearch(self) -> list:
+        filter_query = QueryWithFilters.generate_elasticsearch_query(self.filters, query_type="awards")
+        sort_field = self.get_elastic_sort_by_fields()
+        sorts = [{field: self.pagination["sort_order"]} for field in sort_field]
+        if self.last_id:
+            sorts.append({"generated_unique_award_id.keyword": "desc"})
```

Comment:
so if the user doesn't supply a last_id, we don't sort the results?

Reply:
No, we add sorting by `generated_unique_award_id` because there could be duplicate values for other sort fields and that could result in missing records. The normal sort fields are generated two lines above this in `sort_field = self.get_elastic_sort_by_fields()`

Reply:
Would it be possible to add this functionality to get_elastic_sort_by_fields? As it stands this violates the Single Responsibility Principle and risks logic getting lost.

Reply:
sure